tests/dragonfly/conftest.py

```diff
@@ -229,42 +230,65 @@ def memcached_connection(df_server: DflyInstance):
 
 
 @pytest.fixture(scope="session")
-def gen_tls_cert(df_factory: DflyInstanceFactory):
-    tls_server_key_file_name = "df-key.pem"
-    tls_server_cert_file_name = "df-cert.pem"
+def gen_ca_cert(df_factory: DflyInstanceFactory):
     dfly_path = df_factory.dfly_path
     # We first need to generate the tls certificates to be used by the server
 
-    # Step 1
     # Generate CA (certificate authority) key and self-signed certificate
     # In production, CA should be generated by a third party authority
     # Expires in one day and is not encrtypted (-nodes)
     # X.509 format for the key
     ca_key = dfly_path + "ca-key.pem"
     ca_cert = dfly_path + "ca-cert.pem"
-    step1 = rf'openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout {ca_key} -out {ca_cert} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"'
-    subprocess.run(step1, shell=True)
+    step = rf'openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout {ca_key} -out {ca_cert} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"'
+    subprocess.run(step, shell=True)
+
+    return {"ca_key": ca_key, "ca_cert": ca_cert}
+
 
-    # Step 2
-    # Generate Dragonfly's private key and certificate signing request (CSR)
+@pytest.fixture(scope="session")
+def with_tls_server_args(df_factory: DflyInstanceFactory, gen_ca_cert):
+    dfly_path = df_factory.dfly_path
```

Comment:
actually I think it makes more sense here to use the tmp dir.
df_factory.tmp_dir